lib/include/elements/element/gallery/button.hpp

```diff
@@ -36,6 +36,8 @@ namespace cycfi { namespace elements
 
       enum icon_placement {icon_none, icon_left, icon_right};
       enum label_alignment {align_left, align_center, align_right};
+      enum class direction {up, down, left, right};
+      enum class corner {top_left, top_right, bottom_left, bottom_right};
```

Comment:
I do not use `enum class` for enums in a class. It is redundant and is unnecessarily verbose. The containing class is already a good qualifier.

Also, I think the sequence should also follow the clockwise flow, so: 

```
enum corner {top_left, top_right, bottom_right, bottom_left};
```
Just in case; this can also be used as an index.